### DISCUSSION TITLE
Aufruf des Willy Berg'schen Standort-AT Tools aus WebGIS mit Übergabe von Polygon-Geometrie und weitere Parametern

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#219] [Q&A]
Wir hätten gerne für die Umsetzung des eFWP (elektronischer Flächenwidmungsplan) und der Verwendung des Standort-AT Tools von Willy Berg, die Möglichkeit eines Aufrufes aus dem WebGIS heraus.
Relevant ist, dass die Geometrie vom Benutzer erfasst wurde.

Beispiel: Geoanalyse Report zu einer vom Anwender erfassten Umwidmungsgeometrie.
Der Aufruf würde mittels eines Post-Requests an das Reporting-Tool ausgeführt werden. 
Beispiel Post-Request:
curl -v  -XPOST -d  @test.json -H "content-type: application/json ; charset=utf-8 "  "https://[server]/standort_tool/api/report/pdf/gruppe1/raumanalyse_viia/1/1/31254" --output report.pdf

Inhalt test.json (simple Geometrie)
{"geojson":"{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Polygon\", \"coordinates\":[[[-49000,247000],[-49000,248000],[-48000,248000],[-48000,247000],[-49000,247000]]]}}" }

Wahrscheinlich würde eine Videokonferenz zu dieser Problematik Sinn machen.
Besten Dank!
Andi und Jürgen

**Comment by jugstalt** [comment]
Wir haben bisher sogenannte **benutzerdefinierte Werkzeuge**, bei denen der Anwender in die Karte klickt oder auch eine Geometrie zeichnet und diese dann an eine beliebige Seite weitergeleitet wird, zB euer Höhenprofil. Weiterleitung zu einer anderen Seite ist natürlich ein GET Request.

Für mich stellt sich einmal die Frage, wie das mit einem POST Request funktionieren soll. Theoretisch wäre es schon möglich, dass ich zu irgend einer API etwas hin poste, aber was ist das Ergebnis? Hier ein PDF?

Und soll sowas vom Client (Javascript) überhaupt möglich sein. API sind ja in der Regel in irgend einer weise geschützt (JWT). Am Client kann man die Secrets nur schwer verstecken.

Unser Zugang wäre hier eher so was wie ein **ProcessingServer** Job:
WebGIS ruft ProcessingServer UI auf, übergibt den **Sketch**. Die ``curl`` Abfrage läuft dann am Server (inkl abholen von JWT, etc) und als Ergebnis bekommt der Anwender dann einen Link zum PDF...
